Error in supported operation in README

In this table

```
#>  [1] !=        %%        %/%       &         &&        (         *        
#>  [8] +         -         /         :         <         <-        <=       
#> [15] =         ==        >         >=        Fortran   [         [<-      
#> [22] ^         c         cat       cbind     character declare   double   
#> [29] for       if        ifelse    integer   length    logical   matrix   
#> [36] max       min       numeric   print     prod      raw       seq      
#> [43] sum       which.max which.min {         |         ||
```

These should be changed to
ifelse --> else
seq --> seq_along

Thank you for taking time to look at it and closing my previous issue. 

Unless I am doing something wrong I can't get `ifelse` and `seq` to work. I installed quickr using pak yesterday. Here are some small functions. 

```
library(quickr)

test_ifelse <- function(a, b) {
  declare(
    type(a = double(NA)),
    type(b = double(NA))
  )

  out <- ifelse(a < b, 1, -1)
  out
}

test_seq <- function(a, b) {
  declare(
    type(a = integer(1)),
    type(b = integer(1))
  )

  out <- seq(a, b)
  out
}
```
For ifelse i get the following R error/traceback

```
> q_test_ifelse <- quick(test_ifelse)
Error in `get()`:
! object 'scope' of mode 'function' was not found
Hide Traceback
     ▆
  1. └─quickr::quick(test_ifelse)
  2.   ├─quickr:::compile(r2f(fun))
  3.   │ └─base::stopifnot(inherits(fsub, FortranSubroutine))
  4.   └─quickr:::r2f(fun)
  5.     └─quickr:::new_fortran_subroutine(name, e)
  6.       └─quickr:::r2f(drop_last(body), scope)
  7.         └─quickr (local) handler(...)
  8.           └─base::lapply(args, r2f, scope, ...)
  9.             └─quickr (local) FUN(X[[i]], ...)
 10.               └─quickr (local) handler(...)
 11.                 └─quickr:::r2f(value, scope, ...)
 12.                   └─quickr (local) handler(...)
 13.                     └─base::lapply(args, scope, ...)
 14.                       └─base::match.fun(FUN)
 15.                         └─base::get(as.character(FUN), mode = "function", envir = envir)
```

And for seq I get a compile error

```
> q_test_seq <- quick(test_seq)
using C compiler: ‘gcc (GCC) 15.1.1 20250425 (Red Hat 15.1.1-1)’
using Fortran compiler: ‘GNU Fortran (GCC) 15.1.1 20250425 (Red Hat 15.1.1-1)’
gfortran  -fpic  -g -O2  -c  test_seq_fsub.f90 -o test_seq_fsub.o
gcc -std=gnu23 -I"/opt/R/4.5.0/lib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c test_seq_c_wrapper.c -o test_seq_c_wrapper.o
test_seq_c_wrapper.c: In function ‘test_seq_’:
test_seq_c_wrapper.c:36:30: error: ‘NA’ undeclared (first use in this function); did you mean ‘NAN’?
   36 |   const R_xlen_t out__len_ = NA;
      |                              ^~
      |                              NAN
test_seq_c_wrapper.c:36:30: note: each undeclared identifier is reported only once for each function it appears in
make: *** [/opt/R/4.5.0/lib/R/etc/Makeconf:202: test_seq_c_wrapper.o] Error 1
List of 1
 $ status: int 1
Error in `compile()`:
! Compilation Error
Hide Traceback
    ▆
 1. └─quickr::quick(test_seq)
 2.   └─quickr:::compile(r2f(fun))
```
The generated fortran code looks like

```
subroutine test_seq(a, b, out) bind(c)
  use iso_c_binding, only: c_int
  implicit none

  ! manifest start
  ! args
  integer(c_int), intent(in) :: a
  integer(c_int), intent(in) :: b
  integer(c_int), intent(out), allocatable :: out(:)

  ! locals
  integer(c_int) :: tmp1_
  ! manifest end


  out = [ (tmp1_, tmp1_ = a, b, sign(1, b-a)) ]
end subroutine
```
Claude tells me that this looks like correct fortran code, but there is some problem with the binding to C code.

--- a/R/r2f.R
+++ b/R/r2f.R
@@ -471,11 +471,11 @@ r2f_handlers[["seq"]] <- function(args, scope, ...) {
 
 
 r2f_handlers[["ifelse"]] <- function(args, scope, ...) {
-  .[mask, tsource, fsource] <- lapply(args, scope, ...)
+  .[mask, tsource, fsource] <- lapply(args, r2f, scope, ...)
   # (tsource, fsource, mask)
   mode <- tsource@value@mode
-  dims <- conform(tsource@value, fsource@value)@dims
-  Fortran(glue("merge({tsource}, {fsource}, {mask}"),
+  dims <- conform(mask@value, tsource@value, fsource@value)@dims
+  Fortran(glue("merge({tsource}, {fsource}, {mask})"),
           Variable(mode, dims))
 }
 

--- a/tests/testthat/test-r2f.R
+++ b/tests/testthat/test-r2f.R
@@ -795,3 +795,30 @@ test_that("complex unary intrinsics", {
   }
 
 })
+
+
+test_that("ifelse", {
+  fn <- function(a, b) {
+    declare(
+      type(a = integer(n)),
+      type(b = integer(n))
+    )
+
+    out <- ifelse(a < b, 1, -1)
+    out
+  }
+  expect_quick_identical(fn, list(-10:10, integer(21)))
+
+  # double version of above
+  fn <- function(a, b) {
+    declare(
+      type(a = double(n)),
+      type(b = double(n))
+    )
+
+    out <- ifelse(a < b, 1, -1)
+    out
+  }
+  expect_quick_equal(fn, list(seq(-5, 5, length.out = 20), double(20)))
+
+})

--- a/tests/testthat/test-r2f.R
+++ b/tests/testthat/test-r2f.R
@@ -822,3 +822,18 @@ test_that("ifelse", {
   expect_quick_equal(fn, list(seq(-5, 5, length.out = 20), double(20)))
 
 })
+
+test_that("seq", {
+
+  fn <- function(a, b) {
+    declare(
+      type(a = integer(1)),
+      type(b = integer(1))
+    )
+
+    out <- sum(seq(a, b))
+    out
+  }
+
+  expect_quick_identical(fn, list(-1L, 10L), list(10L, -5L))
+})

